include/kernel.h

```diff
@@ -352,6 +352,27 @@ void k_object_access_all_grant(void *object);
 __syscall void *k_object_alloc(enum k_objects otype);
 
 #ifdef CONFIG_DYNAMIC_OBJECTS
+/**
+ * Allocate memory and install as a generic kernel object
+ *
+ * This is a low-level function to allocate some memory, and register that
+ * allocated memory in the kernel object lookup tables with type K_OBJ_ANY.
+ * Initialization state and thread permissions will be cleared. The
+ * returned z_object's data value will be uninitialized.
+ *
+ * Most users will want to use k_object_alloc() instead.
```

Comment:
stupid question, what are the objects not supported by `k_object_alloc` ?

Reply:
1. futexes and sys_mutex, they live in user and not kernel memory
2. thread stacks, as we do not have an aligned allocator
3. net sockets (this patch) as they are a family of objects of no specific size

There is a switch statement in `z_impl_k_object_alloc()` specifically for this

Reply:
This could be better documented by the way, I'll send a separate PR to augment the doxygen.

Reply:
thanks !